src/main/java/org/beanpod/switchboard/controller/EncoderController.java

```diff
@@ -76,30 +74,33 @@
     }
     // return encoder
     return encoder
+        .map(encoderMapper::toEncoderModel)
         .map(ResponseEntity::ok)
         .orElseThrow(() -> new ExceptionType.DeviceNotFoundException(serialNumber));
   }
 
-  @PostMapping
-  public ResponseEntity<EncoderDto> createEncoder(@RequestBody @Valid EncoderDto encoderDto) {
-    if (encoderDto.getOutput().isEmpty()) {
-      throw new ExceptionType.MissingChannelsException(encoderDto.getSerialNumber());
+  @Override
+  public ResponseEntity<EncoderModel> createEncoder(@RequestBody @Valid EncoderModel encoderModel) {
+    if (encoderModel.getOutput().isEmpty()) {
+      throw new ExceptionType.MissingChannelsException(encoderModel.getSerialNumber());
     }
 
     UserEntity user = userDao.findUser(request.getUserPrincipal().getName());
 
     Optional<DeviceDto> deviceOptional =
-        deviceService.findDevice(user, encoderDto.getSerialNumber());
+        deviceService.findDevice(user, encoderModel.getSerialNumber());
     if (deviceOptional.isEmpty()) {
-      throw new ExceptionType.DeviceNotFoundException(encoderDto.getSerialNumber());
+      throw new ExceptionType.DeviceNotFoundException(encoderModel.getSerialNumber());
     }
+    EncoderDto encoderDto = encoderMapper.toEncoderDto(encoderModel);
     encoderDto.setDevice(deviceOptional.get());
     encoderDto.setLastCommunication(Date.from(Instant.now()));
-    return ResponseEntity.ok(encoderDao.save(user, encoderDto));
+    EncoderDto savedEncoderDto = encoderDao.save(user, encoderDto);
+    EncoderModel savedEncoderModel = encoderMapper.toEncoderModel(savedEncoderDto);
+    return ResponseEntity.ok(savedEncoderModel);
   }
 
-  @DeleteMapping("/{serialNumber}")
-  @Transactional
+  @Override
   public ResponseEntity<String> deleteEncoder(@PathVariable String serialNumber) {
```

Comment:
do you know if we were allowed to delete a decoder that has a channel? It gives an error but i think thats the expected output, just making sure.

Reply:
That's a bug too, good catch! It would've been bad if that went on master.